### ISSUE TITLE
In GSN example, use one-hot cross-entropy (or nll) as classification cost

### ISSUE THREAD

**Issue by lamblin** [lisa-lab/pylearn2#426]
The supervised GSN demonstrated in `pylearn2/scripts/gsn_example.py:test_train_supervised` uses a softmax as cost layer, but `MeanBinaryCrossEntropy` as a cost function. This cost function considers all components of the softmax as independent binomials, which is not correct.

A `OneHotCrossEntropy` cost could be defined, or `pylearn2.costs.supervised_cost.NegativeLogLikelihood` could be made GSN-friendly.

See https://github.com/lisa-lab/pylearn2/pull/392#discussion-diff-6417478
